src/gui/guiKeyChangeMenu.cpp

```diff
@@ -107,16 +107,17 @@ void GUIKeyChangeMenu::regenerateGui(v2u32 screensize)
 	removeAllChildren();
 	key_used_text = nullptr;
 
-	const float s = m_gui_scale;
+	scaling_info info = getScalingInfo(v2u32(835, 430), screensize);
```

Comment:
Since the pattern of

```c++
scaling_info info = getScalingInfo(v2u32(835, 430), screensize);
Comment
	const float s = info.scale;
	const v2u32 size = info.getSize();
	DesiredRect = core::rect<s32>(
		screensize.X / 2 - size.X / 2,
		screensize.Y / 2 - size.Y / 2,
		screensize.X / 2 + size.X / 2,
		screensize.Y / 2 + size.Y / 2
	);
```

is recurring, why not extract this into a function? Something like `core::rect<s32> getCenteredRectForScreensize(v2u32 desired_dimensions, v2u32 screensize)`.

Then you could get rid of the `getScalingInfo` function and the `scaling_info` struct along with it. It would also make the code a bit shorter overall.

Reply:
Indeed. I was already planning to do this, except for the part where we get rid of the `scaling_info` struct. That won't work because the new function still needs to return a scaling factor *and* a rect, which is why I added the struct in the first place.